oncopipe/oncopipe/__init__.py

```diff
@@ -207,24 +192,80 @@ def coerce_namedlist_to_string(obj):
     else:
         dest_dir, dest_file = os.path.split(dest)
     os.makedirs(dest_dir, exist_ok=True)
-
     dest = os.path.join(dest_dir, dest_file)
+    
+    return src, dest
+
+def compare_links(src, dest, overwrite):
+    # Check if the destination file exists and is a symlink
     if os.path.lexists(dest) and os.path.islink(dest):
-        if os.path.realpath(src) == os.path.realpath(dest):
-            return
+        # If the destination link exists and has the same src, exit
+        if src == os.readlink(dest):
+            exit(0) 
         elif overwrite:
             os.remove(dest)
```

Comment:
You probably don't want to use `exit(0)`, which (~I think~ I confirmed this) would cause the Python process to terminate (rather than just exiting the function). In other words, I expect this to cause the Snakefile to stop running rather than skipping the symlinking. 

Essentially, this conditional was intended to skip the call to `os.symlink()`, which raises an error if the destination already exists with the same source. Originally, I just used `return` to cause the `relative_symlink()` function to end early. However, now that this conditional is within a utility function, that's no longer possible. One workaround is to do the same as if `overwrite=True`, which is to remove the symlink knowing that the source won't change. Hence, you can simplify this if-else block to something like this: 

```suggestion
        if src == os.readlink(dest) or overwrite:
            os.remove(dest)
```